### DISCUSSION TITLE
ESP NOW

### DISCUSSION THREAD

**Discussion by JL1306** [micropython/micropython#10863] [ESP32]
Why isn't ESP NOW by Glenn20 incorporated in standard firmware for Micropython for ESP32 and ESP8266?

Glenn has done a lot of great work on this and it would be very usefull if it was included in standard firmware and daily builds, so that people with limited experience, time and resources in building firmware from source can download up to date firmware that includes ESP NOW.

**Comment by davefes** [comment]
I can only guess that there must be some limitations on how many features can be crammed into flash/ram etc.  Fortunately, Glenn provides current images and with ESPNow/WiFi.  I use it in about 1/2 of my projects.

I think it would be appropriate to have a separate category for it on this forum.

**Comment by glenn20** [comment]
Circuitpython have ported the mcropython ESPNOW module into the mainline, so we might see the functionality availabler in production there. Hopefully in micropython some time soon too.
